### ISSUE TITLE
Client stop consuming after a certain time on high load

### ISSUE THREAD

**Issue by sneerin** [confluentinc/confluent-kafka-go#453]
Description
===========
Consumer stop consuming with no error:
I had the same issue with nodejs wrapper for rd cafka driver, I switched to kafkajs on consumer side, producer remains nodejs wrapper. Everything works perfectly. Recently we rolled out one of the services with this driver and having same issue.  After a certain time with high load driver stopped consuming. I have several errors but as far as I get it should be recoverable. 
errors:
`% Error: Local: Broker transport failure: kafka-0.kafka-headless.default:9092/0: Disconnected (after 1798470ms in state UP)`

But after that, it recovered and I kept receiving such messages : 

``2020-04-09 21:57:51.917 EEST
Ignored OffsetsCommitted (<nil>, [activities.list[4]@2929])``

Few other errors : 

`textPayload: "%5|1586442819.782|REQTMOUT|rdkafka#consumer-1| [thrd:GroupCoordinator]: GroupCoordinator/1: Timed out LeaveGroupRequest in flight (after 5263ms, timeout #0): possibly held back by preceeding blocking JoinGroupRequest with timeout in 268137ms
textPayload: "%4|1586442819.782|REQTMOUT|rdkafka#consumer-1| [thrd:GroupCoordinator]: GroupCoordinator/1: Timed out 1 in-flight, 0 retry-queued, 0 out-queue, 0 partially-sent requests
2020-04-09 17:33:39.783 EEST
%3|1586442819.783|ERROR|rdkafka#consumer-1| [thrd:app]: rdkafka#consumer-1: GroupCoordinator: 1 request(s) timed out: disconnect (after 1082013ms in state UP): Local: Timed out
`

How to reproduce
================
No specific steps, on dev env with low data never breaks, on prod after few hours it silently stop consuming.
`	sigchan := make(chan os.Signal, 1)
	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)
	consumerGroup, _ := common.GetEnv("CONSUMER_GROUP")
	brokerURL, _ := common.GetEnv("KAFKA_BROKER")
	c, err := kafka.NewConsumer(&kafka.ConfigMap{
		"bootstrap.servers":               brokerURL,
		"group.id":                        consumerGroup,
		"session.timeout.ms":              10000,
		"broker.address.family":           "v4"})
	if err == nil {
		k.consumer = *c
		info().Msg("Kafka connected")
	} else {
		errorz().Msg("Consumer is not initialized, you probably didn't run bootstrapper")
		return
	}
	keys := reflect.ValueOf(k.subscriptions).MapKeys()
	topics := []string{}
	for k := range keys {
		topics = append(topics, keys[k].Interface().(string))
	}
	//TODO add rebalance options
	k.consumer.SubscribeTopics(topics, nil)

	run := true

	for run == true {
		select {
		//TODO move to Implementation of closer interface
		case sig := <-sigchan:
			errorz().Msgf("Caught signal %v: terminating\n", sig)
			run = false
		default:
			ev := k.consumer.Poll(10)
			if ev == nil {
				continue
			}

			switch e := ev.(type) {
			case *kafka.Message:
				topic := e.TopicPartition.Topic
				go k.subscriptions[*topic](e)

			case kafka.Error:
				// Errors should generally be considered
				// informational, the client will try to
				// automatically recover.
				// But in this example we choose to terminate
				// the application if all brokers are down.
				errorz(err).Msgf(os.Stderr, "%% Error: %v: %v\n", e.Code(), e)
				if e.Code() == kafka.ErrAllBrokersDown {
					run = false
				}
			default:
				debug().Msgf("Ignored %v\n", e)
			}
		}
	}`

Last note, I have  a lot of rebalancing messages with kafkajs, sometimes client is crashing due to non-Kafka reasons or just been killed by Kubernetes due to load.

kafkajs logs:

`[Connection] Response Heartbeat(key: 12, version: 1)
[Runner] The group is rebalancing, re-joining`
So maybe I miss something on rebalancing is there any manual ? This restart behavior doesn't sound likes something extraordinary for the high load cluster. Any debug I can enable to trace the error? 

Checklist
=========
Please provide the following information:

 - [x ] confluent-kafka-go and librdkafka version: (`librdkafka@edgecommunity>=1.3.0`) 16974079 1.3.0
 - [x ] Apache Kafka broker version: 2.0.1 ( confluent 5.0.1) 
 - [x ] Client configuration: `&kafka.ConfigMap{
		"bootstrap.servers":               brokerURL,
		"group.id":                        consumerGroup,
		"session.timeout.ms":              10000,
		"broker.address.family":           "v4"}`
 - [x ] Operating system: linux
 - [ x] Provide client logs (with `"debug": ".."` as necessary)
 - [x ] Provide broker log excerpts
 - [x] Critical issue

**Comment by sneerin** [comment]
I can't reproduce it again so far. closing.

**Comment by edenhill** [comment]
Since you're spinning up a go-routine for each received message, could it be that the Go scheduler starves the consume thread (your main loop here) so long that max.poll.interval.ms kicks in and evicts the member from the group? Are there any such logs?

What kind of instance/hardware is this running on? How many vcpu/cores?

Also see the release notes for librdkafka v1.4.0, there are a couple of related consumer fixes:
https://github.com/edenhill/librdkafka/releases/tag/v1.4.0

**Comment by sneerin** [comment]
I updated the version, it's not repeating for 2 weeks already, despite the load, so it seems good. 
The resources are not an issue, it's running on GKE with 6 shared vcpu's. per node.  
Regarding the go-routine seems like it's fine, the message handler is quite heavy, so I moved it to separate goroutine. 
No specific errors anymore. 
We are happy with migration, planning more microservices using this stack :) 
Thanks a lot for the support ! :)
